From 7b2a751b301f4066fc392c0041d3163813f4fa10 Mon Sep 17 00:00:00 2001
From: Carlos Alves <cbc.alves@gmail.com>
Date: Wed, 5 May 2021 06:43:41 -0300
Subject: [PATCH] Fix Screen::_cuY position inside screen

Keep _cuY position inside max screen lines in AppScreen mode.
---
 src/Screen.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Screen.cpp b/src/Screen.cpp
index 666d63cd..a9b48cc3 100644
--- a/src/Screen.cpp
+++ b/src/Screen.cpp
@@ -419,6 +419,7 @@ void Screen::setCursorLine(int newLine)
 {
     if (_currentModes[MODE_AppScreen] == 1) {
         _savedState.cursorLine = newLine;
+        _cuY = qBound(0, _cuY, _lines - 1);
     } else {
         _cuY = newLine;
     }
-- 
GitLab

